sympy/integrals/laplace.py

```diff
@@ -644,15 +645,14 @@ def _laplace_trig_expsum(f, t):
             xm.append({'k': term, 'a': 0, re: 0, im: 0})
             continue
         term = term.powsimp(combine='exp')
-        if (r := term.match(p*exp(m))) is not None:
-            if (mp := r[m].as_poly(t)) is not None:
-                mc = mp.all_coeffs()
-                if len(mc) == 2:
-                    xm.append({
-                        'k': r[p]*exp(mc[1]), 'a': mc[0],
-                        re: re(mc[0]), im: im(mc[0])})
-                else:
-                    xn.append(term)
+        if (
+                (r := term.match(p*exp(m))) is not None
+                and (mp := r[m].as_poly(t)) is not None):
+            mc = mp.all_coeffs()
+            if len(mc) == 2:
+                xm.append({
+                    'k': r[p]*exp(mc[1]), 'a': mc[0],
+                    re: re(mc[0]), im: im(mc[0])})
             else:
                 xn.append(term)
```

Comment:
Neither of these is great but I think that the nested if is easier to read. Weird black-style formatting of complicated conditions is just a sign that something else needs to change.

I'd write this more like:
```python
if (r := term.match(p*exp(m))) is not None:
    if (mp := r[m].as_poly(t)) is not None:
        coeffs = mp.all_coeffs()
        if len(coeffs) == 2:
            c1, c2 = coeffs
            params = {'k': r[p]*exp(c2), 'a': c1, re: re(c1), im: im(c1)}
            xm.append(params)
        else:
            xn.append(term)
```
Then again I'm not sure why we need the nested matching and poly checking. Here we only accept a polynomial of two terms so can it not be part of the match?
```python
if (r := term.match(p*exp(c1*t + c2)) is not None):
```

Reply:
My problem is that this will not match

```
f = exp((t+a)*(t+b)-t**2)
```

I have to somehow deal with it.

Reply:
I now have a good solution for this, I think. I made a small modification to `_laplace_deep_collect`, and now it serves the purpose.

Reply:
One more change has happened, now it is completely ready for review.